app/src/main/kotlin/dogacel/kotlinx/protobuf/gen/TypeNames.kt

```diff
@@ -71,7 +71,7 @@ object TypeNames {
             }
 
         if (
-            fieldDescriptor.hasOptionalKeyword() ||
+            fieldDescriptor.isOptional ||
```

Comment:
I think this causes a bug, we should update this to be `unconditionallyHasValue` as well.

Reply:
Or maybe `hasFieldPresence()` is enough. Let's give it a try.